src/application.rs

```diff
@@ -29,7 +29,7 @@ pub struct HttpApplication<S = ()> {
 #[doc(hidden)]
 pub struct Inner<S> {
     prefix: usize,
-    default: ResourceHandler<S>,
+    default: Rc<RefCell<ResourceHandler<S>>>,
```

Comment:
Clone would be unexpected. Problem is Handler trait uses mut self. Maybe we should remove it from Handler. Then we can remove all RefCells

Reply:
Ah... then maybe yes it would be better to not make it `&mut`
In first place it should not be mutable during routing so maybe we should first make it immutable and then go without `RefCell` especially since it implies runtime cost